config/initializers/highlight_override.rb

```diff
@@ -0,0 +1,54 @@
+# Monkey patch the generated Bindings for the swagger Highlight model
+
+Api::V0::Bindings::NewHighlight.class_exec do
+  def valid_location_strategies?
+    location_strategies.present? &&
+      location_strategies.none? { |item| item.nil? } &&
+      location_strategies.all?(&:valid?)
+  end
+
+  def location_strategies=(array)
+    @location_strategies = array.map do |item|
+      case item[:type]
+      when 'TextPositionSelector'
+        Api::V0::Bindings::TextPositionSelector.new(item)
+      when 'XpathRangeSelector'
+        Api::V0::Bindings::XpathRangeSelector.new(item)
+      end
+    end
+  end
+
+  alias_method :old_valid?, :valid?
+  def valid?
+    old_valid? && valid_location_strategies?
+  end
+
+  alias_method :old_list_invalid_properties, :list_invalid_properties
+  def list_invalid_properties
+    invalid_properties = old_list_invalid_properties
+
+    if location_strategies.blank?
+      invalid_properties.push('invalid value for "location_strategies", location_strategies cannot be empty.')
+    end
+
+    if location_strategies.any? { |item| item.nil? }
```

Comment:
this one will blow up if the array is nil.  you catch this below with the `&.` — maybe would be good to move this check into that next loop since we also check there for `strategy.nil?`?